### PR TITLE
Theme improvements

### PR DESCRIPTION
**Pull request by nhunzaker**
This PR updates our themes to better match the original designs. Along the way, I made a few non-visual updates too:

- Broke out player states into separate components
- Introduced the idea of an intro, scene, and end state
- Removed legacy slant button

The themes aren't perfect, but this is an important step forward for writing new themes.

---

image

image

image

screenshot 2019-01-10 at 9 56 56 am

### CODE REVIEW

File: app/javascript/src/components/Player.tsx [hunk 1]
```diff
@@ -37,6 +49,8 @@ class Player extends React.Component<PlayerProps, PlayerState> {
 
     this.engine.installDefaultFactories()
     this.model.deSerializeDiagram(this.props.story, this.engine)
+
+    import(`./themes/${props.theme}Theme.css`)
```
Comment (leobauza) [inline app/javascript/src/components/Player.tsx]: could u use this to delay rendering or something like that on this (not sure how the thing all works so maybe not necessary) just always see this kind of import then utilized as a promise

File: app/javascript/src/components/PlayerInvalid.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react'
+
+export const PlayerInvalid: React.SFC<{}> = () => {
+  return (
+    <main>
+      Your story's all messed up. You probably deleted the start scene.
+      <br />
+      <br />
+      <b>Shouldn't have done that.</b>
```
Comment (leobauza) [inline app/javascript/src/components/PlayerInvalid.tsx]: `<p>` and `<strong>` here maybe?

File: app/javascript/src/components/Player.tsx (hunk 1 above)
Comment (nhunzaker) [reply app/javascript/src/components/Player.tsx]: Ya. We definitely could. I should take a second pass at this. These could just as easily be included in the head of the document.

File: app/javascript/src/components/PlayerInvalid.tsx (hunk 2 above)
Comment (nhunzaker) [reply app/javascript/src/components/PlayerInvalid.tsx]: This markup is rough... Yes.

Comment (efatsi) [reply app/javascript/src/components/PlayerInvalid.tsx]: you talking trash about my markup???

Comment (nhunzaker) [reply app/javascript/src/components/PlayerInvalid.tsx]: @efatsi it was a group effort. Let's be trash friends.

**Comment by efatsi** [comment]
👍 👍

File: app/javascript/src/components/PlayerInvalid.tsx (hunk 2 above)
Comment (nhunzaker) [reply app/javascript/src/components/PlayerInvalid.tsx]: Ack. I didn't do this... Sorry @leobauza.

PR Shortly.

**Pull request merged**